sdk/elixir/runtime/main.go

```diff
@@ -0,0 +1,177 @@
+	ctx context.Context,
+	modSource *ModuleSource,
+	introspectionJson string,
+) (*Container, error) {
+	ctr, err := m.CodegenBase(ctx, modSource, introspectionJson)
+	if err != nil {
+		return nil, err
+	}
+
+	modName, err := modSource.ModuleName(ctx)
+	if err != nil {
+		return nil, err
+	}
+	subPath, err := modSource.SourceSubpath(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	mod := normalizeModName(modName)
+	entrypoint := path.Join(ModSourceDirPath, subPath, mod)
+
+	return ctr.
+		WithEntrypoint([]string{
+			"mix", "cmd",
+			"--cd", entrypoint,
+			"mix do deps.get + dagger.invoke",
+		}), nil
+}
+
+func (m *ElixirSdk) Codegen(
+	ctx context.Context,
+	modSource *ModuleSource,
+	introspectionJson string,
+) (*GeneratedCode, error) {
+	ctr, err := m.CodegenBase(ctx, modSource, introspectionJson)
+	if err != nil {
+		return nil, fmt.Errorf("could not load module config: %v", err)
+	}
+
+	return dag.GeneratedCode(ctr.Directory(ModSourceDirPath)).
+		WithVCSGeneratedPaths([]string{genDir + "/**"}).
+		WithVCSIgnoredPaths([]string{"dagger"}), nil
+}
+
+func (m *ElixirSdk) CodegenBase(
+	ctx context.Context,
+	modSource *ModuleSource,
+	introspectionJson string,
+) (*Container, error) {
+	modName, err := modSource.ModuleName(ctx)
+	if err != nil {
+		return nil, err
+	}
+	subPath, err := modSource.SourceSubpath(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	mod := normalizeModName(modName)
+
+	codegenDepsCache, codegenBuildCache := mixProjectCaches(dag, "dagger-codegen")
+
+	ctr := m.Base("").
+		WithMountedDirectory(ModSourceDirPath, modSource.ContextDirectory()).
+		WithMountedDirectory(sdkSrc, m.SDKSourceDir).
+		WithMountedCache(codegenPath()+"/deps", codegenDepsCache).
+		WithMountedCache(codegenPath()+"/_build", codegenBuildCache).
+		With(installCodegen).
+		WithNewFile(schemaPath, ContainerWithNewFileOpts{
+			Contents: introspectionJson,
+		}).
+		WithWorkdir(path.Join(ModSourceDirPath, subPath)).
+		WithDirectory(
+			"dagger",
+			m.SDKSourceDir,
+			ContainerWithDirectoryOpts{Exclude: []string{
+				"*.livemd",
+				"*.md",
+				".changes",
+				"dagger_codegen",
+				"runtime",
+				"scripts",
+				"test",
+			}},
+		).
+		WithWorkdir(path.Join(ModSourceDirPath, subPath, "dagger")).
+		WithExec([]string{
+			"dagger_codegen", "generate",
+			"--outdir", "lib/dagger/gen",
+			"--introspection", schemaPath,
+		}).
+		WithExec([]string{
+			"mix", "format",
+		}).
+		WithWorkdir(path.Join(ModSourceDirPath, subPath))
+
+	// Generate scaffolding code when no project exists.
+	if _, err = ctr.Directory(mod).File("mix.exs").Sync(ctx); err != nil {
+		ctr := ctr.
+			WithExec([]string{"mix", "new", "--sup", mod}).
+			WithExec([]string{"mkdir", "-p", mod + "/lib/mix/tasks"}).
+			WithExec([]string{"elixir", "/sdk/runtime/template.exs", "generate", mod})
+
+		return ctr, nil
+	}
+	return ctr, nil
+}
+
+func (m *ElixirSdk) Base(version string) *Container {
+	if version == "" {
+		version = defaultElixirVersion
+	}
```

Comment:
Don't see a need for acepting the `version` as function argument.

Reply:
You're right. I think we can remove it for now.

Reply:
Fixed in https://github.com/dagger/dagger/pull/6967/commits/a644d96ed3a67dbc2dac46ad64bfd39bcaa32acf